addons/payment_test/models/payment_transaction.py

```diff
@@ -0,0 +1,25 @@
+# coding: utf-8
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import logging
+
+from odoo import models
+
+_logger = logging.getLogger(__name__)
+
+
+class PaymentTransaction(models.Model):
+    _inherit = 'payment.transaction'
+
+    def _send_payment_request(self):
+        """ Request our Imaginary acquirer to execute the payment.
+
+        :return: None
+        """
+        super()._send_payment_request()  # Log the 'sent' message
+        if self.acquirer_id.provider != 'test':
+            return
+
+        # The token payment test is always a success.
+        self._set_done()
+        self._execute_callback()
```

Comment:
Instead of this, prefer making a call to the `/payment/test/payment` route to obtain the same result while fully emulating the standard flow.

Reply:
To avoid making request to localhost, I prefer to
`self._handle_feedback_data('test', {'reference': self.reference})`
from the _send_payment_request method.